scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -289,21 +289,22 @@ class ScalametaParser(input: Input)(implicit dialect: Dialect) { parser =>
       if (!dialect.allowSignificantIndentation) false
       else {
         val existingIndent = sepRegions.find(_.isIndented).map(_.indent).getOrElse(0)
-        val hasLF = (prevPos to tokenPos).exists(i => scannerTokens(i).is[LF])
+        val lastLF = (prevPos to tokenPos).reverse.collectFirst({
```

Comment:
i'd short circuit, perhaps, and avoid reverse search if the condition is false
```
expected < existingIndent && .... collectFirst {
  case pointPos if ... =>
    curr = TokenRef ...
}.isDefined

Reply:
thx! I'll look at it in the next PR